tls/options.go

```diff
@@ -0,0 +1,159 @@
+package tls
+
+import (
+	"context"
+	stdtls "crypto/tls"
+	"crypto/x509"
+	"os"
+	"time"
+
+	rbacproxytls "github.com/brancz/kube-rbac-proxy/pkg/tls"
+	"github.com/go-kit/log"
+	"github.com/oklog/run"
+)
+
+// UpstreamOptions represents the options of the upstream TLS configuration
+// this structure contains the certificates and the watchers if the certificate/ca watchers are enabled.
+type UpstreamOptions struct {
+	cert         *stdtls.Certificate
+	ca           []byte
+	certReloader *rbacproxytls.CertReloader
```

Comment:
I think we could improve the name here to try to make things easier to understand. `*rbacproxytls.CertReloader` is a certificate reloader for the tenants MTLS certs. So maybe `UpstreamOptions.certReloader` could be named `UpstreamOptions.tenantsCertReloader`.

Without this it makes me wonder: why there is a `certReloader` and a `caReloader` here? Both are certs!

This brings me to wonder about the name of the struct itself. What is "upstream" in `UpstreamOptions`? 🤔

Reply:
why `tenantsCertReloader`? I don't see where the tenants are involved here. Upstream means all upstream services for logs/metrics/traces.  At least is the terminology that is used in all the gateway. for instance:

`    	File containing the TLS client key to authenticate against upstream logs servers. Leave blank to disable mTLS.`

Reply:
Humm, I see. So I got it wrong. Probably we could clarify this. There are many certs everywhere, and reloaders for them, so things are confusing.